### DISCUSSION TITLE
Datetime being exported with ",000"

### DISCUSSION THREAD

**Discussion by codegardenbr** [DaxStudio/DaxStudio#654] [Q&A]
Hi guys,

Datetime are being exported to CSV with ",000" in the end, altough it doesnt show that in PowerBI or even Grid result in Dax Studio. I heard that I should with the query, but I'm new in this. Can you guys tell me whats wrong?

/* START QUERY BUILDER */
EVALUATE
SUMMARIZECOLUMNS(
    TCD_ORDER_INVOICE_ITEM[ID],
    TCD_ORDER_INVOICE_ITEM[ID_ORDER_INVOICE],
    TCD_ORDER_INVOICE_ITEM[ID_ORDER_ITEM],
    FORMAT(TCD_ORDER_INVOICE_ITEM[DATE_CREATED], "yyyy-MM-dd HH:mm:ss"),
    TCD_ORDER_INVOICE_ITEM[QUANTITY],
    TCD_ORDER_INVOICE_ITEM[FLAG_ACTIVE]
)
/* END QUERY BUILDER */

**Comment by dgosbell** [comment]
That does not look like a valid DAX query. Was that the actual query you used to extract your data as you are converting the date to string so it should export exactly as specified?

How are you exporting the data? Are you using the output to file option or the Export Data option in the Advanced tab?

Can you run a query against the Adventure Works example pbix file linked in the tutorial page here https://daxstudio.org/tutorials/writing-dax-queries/ and attach an example of the output issue you are describing?

**Reply by codegardenbr** [reply]
Hello,

Dax Studio is helping me to export data from PowerBI into CSV to dataload in Oracle from, SQL Oracle Developer Wizard. 

I'm running the query in query builder, Home tab. Everything looks amazing. Lucky to have Dax Studio available.

If I maintain the current query:

/* START QUERY BUILDER */
EVALUATE
SUMMARIZECOLUMNS(
    TCD_ORDER_DEALER_INVOICE_ITEM[ID],
    TCD_ORDER_DEALER_INVOICE_ITEM[ID_ORDER_DEALER_INVOICE],
    TCD_ORDER_DEALER_INVOICE_ITEM[ID_ORDER_DEALER_ITEM],
    TCD_ORDER_DEALER_INVOICE_ITEM[DATE_CREATED],
    TCD_ORDER_DEALER_INVOICE_ITEM[QUANTITY],
    TCD_ORDER_DEALER_INVOICE_ITEM[FLAG_ACTIVE]
)
/* END QUERY BUILDER */

It works as it should in Grid result panel. With the date formated as "2016-12-05 13:57:36".

When I export to CSV, tha same date exports as "2016-12-05 13:57:36,000". Notice the ",000" in the end.

So, this is an issue for me, because I have to open all files with Notepad++ and replace ",000" with "" (blank). Oracle developer is bringing me an error because of that ",000".

So I Googled and found a forum post somewhere, from a guy with the same issue, and they recommend to format the date in the query.

**Comment by dgosbell** [comment]
If you open these files in Excel do they open correctly? I am using a standard ISO8601 datetime format and the trailing ,000 is the fractional sections exported using the regional settings from your PC. (so US/UK will use a period for the decimal separator instead of a comma).

I'm not sure why Oracle Developer would have an issue dealing with this format as it's fairly standard. However I think most of the time most people probably do not store fractions of seconds in their date time columns so I might see if I can change the default format to exclude the milliseconds portion of datetimes or provide an option for this.
